simpleui-server/test/ZMQ.spec.ts

```diff
@@ -0,0 +1,119 @@
+var _zmq = require('zeromq');
+import { ZMQ_Socket_Wrapper } from '../src/sui_zmq';
+
+
+//import { Logger, LogLevel } from '../src/server-logger';
+// setup logger
+//Logger.setLogLevel("DEBUG");
+
+
+class ZMQ_Reply_Socket {
+    socket: any
+    port: number;
+    address: string;
+
+    constructor(port: number) {
+        try {
+            this.port = port;
+            this.address = `tcp://127.0.0.1:${this.port}`;
+            this.socket = _zmq.socket('rep');
+            this.socket.setsockopt(_zmq.ZMQ_LINGER, 0);
+            this.socket.bindSync(this.address);
+
+        } catch (err) {
+           console.error(`Could not make REPLY socket on ${this.address}, got error: ${err}`);
+        }
+    }
+    send(msg: string) {
+        try {
+            this.socket.send(msg);
+        } catch (err) {
+            console.error(`Could not send zmq message: ${msg} got error: ${err}`);
+        }
+    }
+    close() {
+        try {
+            if (this.socket.closed === false) {
+                this.socket.close();
+                console.info(`Socket closed ${this.address}`);
+            }
+        } catch (err) {
+            console.error(`Could not close socket ${this.address}, got error: ${err}`);
+        }
+    }
+}
+
+
+
+
+
+const PORT = 1234;
+const HOSTNAME = '127.0.0.1';
+let mock_reply_socket: ZMQ_Reply_Socket;
+let test_request_socket: ZMQ_Socket_Wrapper;
+
+
+
+
+
+describe('ZMQ_Socket_Wrapper Class testing', () => {
+
+    beforeEach( () => {
+        test_request_socket = new ZMQ_Socket_Wrapper(HOSTNAME, PORT);
+        mock_reply_socket = new ZMQ_Reply_Socket(PORT);
+    });
+    afterEach( () => {
+        test_request_socket.close();
+        mock_reply_socket.socket.unbindSync(mock_reply_socket.address);
+        mock_reply_socket.close();
+    });
+
+
+    // class inits properly
+    test('Class instance exist', () => {
+        expect(test_request_socket).toBeTruthy();
+    });
+
+
+    // also tests that sockets connect on init (bc of beforeEach())
+    test('REQ socket should recieve message from REP socket', done => {
+        const test_message = "ABCDEFG";
+        test_request_socket.send(test_message);
+        mock_reply_socket.socket.on('message', (raw_data: any) => {
+            const zmq_msg = raw_data.toString();
+            expect(zmq_msg).toEqual(test_message);
+            done();
+        });
+    });
+
+
+    // times out occasionally because the socket from the previous test
+    // hasnt closed yet and therefore cannot create a new socket
```

Comment:
presumably you're still working on these?

Reply:
oops! Sorry that was an old comment. I fixed those tests to that issue doesnt happen anymore